batch/batch/cloud/azure/driver/regions.py

```diff
@@ -20,6 +21,8 @@ def choose_location(
         local_ssd_data_disk: bool,  # pylint: disable=unused-argument
         data_disk_size_gb: int,  # pylint: disable=unused-argument
         preemptible: bool,  # pylint: disable=unused-argument
-        region: Optional[str],  # pylint: disable=unused-argument
+        regions: List[str],
     ) -> str:
-        return region or self._default_region
+        if self._default_region not in regions:
+            raise RegionsNotSupportedError(regions, [self._default_region])
+        return self._default_region
```

Comment:
This feels like a departure from the current behavior, which would be more like `random.choice(regions) if regions else self._default_region`. It looks to me here like since we only support one region in azure (though I'm not exactly sure why), you want to limit to just that region. I'm ok with doing that for now, but in that case I think we should raise an error if regions is not exactly equal to `[self._default_region]`.